test/parallel/test-fs-realpath.js

```diff
@@ -201,9 +201,11 @@ function test_cyclic_link_protection(callback) {
     fs.symlinkSync(t[1], t[0], 'dir');
     unlink.push(t[0]);
   });
-  assert.throws(function() { fs.realpathSync(entry); });
+  assert.throws(function() { fs.realpathSync(entry); }, /^Error: ELOOP/);
```

Comment:
One downside to `common.expectsError()` is that it increases the abstractions people have to learn about in order to work with even simple tests. Whereas before, all they had to know about is `assert.throws()`, now they have to *also* know about `common.expectsError()`. This is very different (IMO) from `common.mustCall()` in that the latter has an intuitively understandable name, accepts arguments as one would expect, and (in most cases) doesn't actually require reading documentation or code to figure out what it's doing. With `common.expectsError()`, there's a fair bit of magic. Like, it's not obvious what the first argument would be. Or the second. Or the third. You just have to know.

For that reason, I'm inclined to accept (encourage even) people submitting first PRs that add a regexp as a second argument to `assert.throws()` calls that previously only had a single argument. Changing the regexp to `common.expectsError()` can be their second PR or a subsequent PR from someone else.

Another thing that might help is if the arguments accepted by `common.expectsError()` was a single settings object. Someone coming across this has a hope of understanding what's going on without reading source or docs:

```js
const validatorFunction = common.expectsError({code: 'ELOOP', 
                                               constructor: Error,
                                               message: 'looped doodad found'});
```

This, by comparison, is harder to grok without referring to docs or code. And only people who work with tests frequently can be expected to remember the three arguments and their order. By comparison, remembering that the error code is `code` and the message is `message` might be manageable.

```js
const validatorFunction = common.expectsError('ELOOP', 
                                               Error,
                                               'looped doodad found');